### DISCUSSION TITLE
How to make actions available when only relative items are selected for this action ?

### DISCUSSION THREAD

**Discussion by PetroGromovo** [filamentphp/filament#6460] [Help] [answered]
Looking how actions works at https://filamentphp.com/docs/2.x/tables/actions

I made 2 actions to Publish/UnPublish items:

```
            ->bulkActions(
                [
                    Tables\Actions\DeleteBulkAction::make(),
                    BulkAction::make('Publish')
                        ->label('Publish selected')
                        ->action(function (Collection $records, array $data): void {
                            foreach ($records as $itemRecord) {
                                $itemRecord->published = true;
                                $itemRecord->save();
                            }
                        })
                        ->deselectRecordsAfterCompletion(),

                    BulkAction::make('Unpublish')
                        ->label('Unpublish selected')
                        ->action(function (Collection $records, array $data): void {
                            foreach ($records as $itemRecord) {
                                $itemRecord->published = false;
                                $itemRecord->save();
                            }
                        })
                        ->deselectRecordsAfterCompletion(),
                ]

```

it works and I wonder if there is a way to have any of actions available when only relative items are selected for this action ?
in other words : when any item is (un)selected possibility to to recalcule that 'Publish' is visible/enabled if there only 'Unpublished'
items()) selected?

I found method isTableRecordSelectable() - but it looks differerently...

```
        "filament/filament": "^2.17.34",
        "laravel/framework": "^9.52.5",

```

Thanks in advance!

**Comment by danharrin** [comment]
For example, `->visible(fn ($record) => ! $record->published)` to show an action only when the record is not published

**Comment by PetroGromovo** [comment]
1) I tried to insert row as you proposed, like :
```
                    BulkAction::make('Publish')
                        ->label('Publish selected')
                        ->visible(fn ($record) => ! $record->published)
                        ->action(function (Collection $records, array $data): void {
                            foreach ($records as $itemRecord) {
                                $itemRecord->published = true;
                                $itemRecord->save();
                            }
                        })

```
But I got error :

`Unable to resolve dependency [Parameter #0 [ <required> $record ]] in class App\Filament\Resources\ItemResource
`

 2) I am still not sure what is
`    fn ($record) => ! $record->published
`in visible condition ?
I need somehow to run in all items in current listing page and check how many unpublished items are selected when user open 
bulk menu items...

**Reply by danharrin** [reply, marked as answer]
You can't use `$record` here to disable it conditionally, The performance wouldn't be good enough on large datasets if we allowed that
